### DISCUSSION TITLE
steps needed to disable authentication requirement?

### DISCUSSION THREAD

**Discussion by gw37** [microsoft/sample-app-aoai-chatGPT#128] [Q&A]
Hi, we're using this app for some rapid prototyping and want to disable the requirement for authentication. If anyone have an example of how this can be done I would appreciate a pointer.  I tried implementing something following the suggestion of "update the logic in getUserInfoList()" but so far that has not worked.  thank you !

**Comment by yasser2652** [comment]
Hi, I noticed your query about disabling the authentication requirement for rapid prototyping. Were you able to find a solution to successfully update the logic in getUserInfoList()? Your insights would be valuable. I am running the 'Basic Chat Experience' code locally and encountering a 'Resource not found" error

**Reply by gw37** [reply]
hi i did not find an easy way to get this done so its still an open issue. i'm hopeful someone will create some guidance about this.

**Comment by sarah-widder** [comment]
Hi folks. You can disable the auth check by updating getUserInfoList like so. Comment out the existing logic and replace it with a single call to setShowAuthMessage(false)
```
const getUserInfoList = async () => {
        setShowAuthMessage(false);

        // const userInfoList = await getUserInfo();
        // if (userInfoList.length === 0 && window.location.hostname !== "127.0.0.1") {
        //     setShowAuthMessage(true);
        // }
        // else {
        //     setShowAuthMessage(false);
        // }
    }
```
You can also set the default auth message state to false on line 49:
    `const [showAuthMessage, setShowAuthMessage] = useState<boolean>(false);`
and comment out or remove the useEffect that calls getUserInfoList around line 494.

**Reply by jkraycfgi** [reply]
Hello Sarah - I tried these changes, and we do have Authorization setup, but when we load the web app, we have the authentication screen blink for half a second for some reason before the chat interface loads. Any thoughts? Thank you.

**Reply by npsquared** [reply]
@jkraycfgi 
Take the following section of code out in Chat.tsx:
```
  {showAuthMessage ? (
                <Stack className={styles.chatEmptyState}>
                    <ShieldLockRegular className={styles.chatIcon} style={{ color: 'darkorange', height: "200px", width: "200px" }} />
                    <h1 className={styles.chatEmptyStateTitle}>Authentication Not Configured</h1>
                    <h2 className={styles.chatEmptyStateSubtitle}>
                        This app does not have authentication configured. Please add an identity provider by finding your app in the
                        <a href="https://portal.azure.com/" target="_blank"> Azure Portal </a>
                        and following
                        <a href="https://learn.microsoft.com/en-us/azure/app-service/scenario-secure-app-authentication-app-service#3-configure-authentication-and-authorization" target="_blank"> these instructions</a>.
                    </h2>
                    <h2 className={styles.chatEmptyStateSubtitle} style={{ fontSize: "20px" }}><strong>Authentication configuration takes a few minutes to apply. </strong></h2>
                    <h2 className={styles.chatEmptyStateSubtitle} style={{ fontSize: "20px" }}><strong>If you deployed in the last 10 minutes, please wait and reload the page after 10 minutes.</strong></h2>
                </Stack>
            )
```

**Comment by sarah-widder** [comment]
We've also added an optional environment variable setting AUTH_ENABLED which can be set to false, which will disable the authentication check. From the README:
<img width="532" alt="image" src="https://github.com/microsoft/sample-app-aoai-chatGPT/assets/19666316/2a8a1092-c86f-4e5d-aa82-2c4e2438a6ab">

**Reply by madushann** [reply]
I have tried to disable by changing configuration AUTH_ENABLED = False
But still it keeps redirecting to login page for authentication. Not sure what I am missing here. 
Any help?
